internal/alloydb/refresh.go

```diff
@@ -113,99 +114,82 @@ func fetchEphemeralCert(
 	}
 	resp, err := cl.GenerateClientCertificate(ctx, req)
 	if err != nil {
-		return certChain{}, errtype.NewRefreshError(
+		return nil, errtype.NewRefreshError(
 			"create ephemeral cert failed",
 			inst.String(),
 			err,
 		)
 	}
-	// There should always be two certs in the chain. If this fails, the API has
-	// broken its contract with the client.
-	if len(resp.PemCertificateChain) != 2 {
-		return certChain{}, errtype.NewRefreshError(
-			"missing instance and root certificates",
+
+	certChainPEM := append([]string{resp.PemCertificate}, resp.PemCertificateChain...)
+	certPEMBlock := []byte(strings.Join(certChainPEM, "\n"))
+	keyPEMBlock := &pem.Block{
+		Type: "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(key),
+	}
+
+	cert, err := tls.X509KeyPair(certPEMBlock, pem.EncodeToMemory(keyPEMBlock))
+	if err != nil {
+		return nil, errtype.NewRefreshError(
+			"create ephemeral cert failed",
 			inst.String(),
 			err,
 		)
 	}
-	rc, err := parseCert(resp.PemCertificateChain[1]) // root cert
-	if err != nil {
-		return certChain{}, errtype.NewRefreshError(
-			"failed to parse root cert",
+
+	// TODO(fixme) Take the root cert from the cert chain for now.
```

Comment:
Where would you expect to get this otherwise?

Reply:
Root cert can be sent separately by the API.